### PR TITLE
[Merged by Bors] - TY-3008 Web service setup

### PR DESCRIPTION
**Pull request by biernacki**
**Summary**:
- created `web-api` crate with warp as the web server
- added initial boilerplate for required routes
- added linters
- added `DE_PORT` and `DE_IP_ADDR` env vars for configuration

**References**:
- [TY-3008]
- [TY-3012]

[TY-3008]: https://xainag.atlassian.net/browse/TY-3008?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[TY-3012]: https://xainag.atlassian.net/browse/TY-3012?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: discovery_engine_core/web-api/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[package]
+name = "web-api"
+version = "0.1.0"
+license = "AGPL-3.0-only"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (janpetschexain) [inline discovery_engine_core/web-api/Cargo.toml:7]: :art: this can be removed

File: discovery_engine_core/web-api/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+[dependencies]
+serde = { version = "1.0.137", features = ["derive"] }
+serde_json = "1.0.81"
+tokio = { version = "1.19.2", features = ["full"] }
```
Comment (janpetschexain) [inline discovery_engine_core/web-api/Cargo.toml:12]: ❓ do we actually need the full `tokio`? if not we can save some space by selecting required features

File: discovery_engine_core/web-api/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+)]
+
+use std::{convert::Infallible, env, net::IpAddr};
+use uuid::Uuid;
+use warp::{hyper::StatusCode, Filter};
+
+#[tokio::main]
+async fn main() {
+    // TODO: TY-3011 - add filepath env var for documents data json file
+    let port = env::var("PORT")
```
Comment (janpetschexain) [inline discovery_engine_core/web-api/src/main.rs:40]: ❓ can we be sure that there are no name conflicts? this env var name is quite general, maybe we should prefix it like `DE_PORT` or sth else. similarly for the `IP_ADDR`.

File: discovery_engine_core/web-api/Cargo.toml (hunk 2 above)
Comment (biernacki) [reply discovery_engine_core/web-api/Cargo.toml]: I've limited features to `macros`, if I understand correctly should be enough for now

**Review by janpetschexain — APPROVED** [review]
lgtm

File: discovery_engine_core/web-api/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+#[tokio::main]
+async fn main() {
+    // TODO: TY-3011 - add filepath env var for documents data json file
+    let port = env::var("DE_PORT")
+        .unwrap_or_else(|_| "3000".to_string())
+        .parse::<u16>()
+        .unwrap();
+    let ip_addr = env::var("DE_IP_ADDR")
+        .unwrap_or_else(|_| "127.0.0.1".to_string())
```
Comment (biernacki) [inline discovery_engine_core/web-api/src/main.rs:45]: change to `0.0.0.0`

File: discovery_engine_core/web-api/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    // PATH /user/:user_id
+    let user_route = warp::path("user").and(warp::path::param::<Uuid>());
+
+    // GET /user/:user_id/documents
+    let ranked_documents = user_route
+        .and(warp::path("documents"))
+        .and(warp::get())
+        .and_then(handle_ranked_documents);
+
+    // POST /user/:user_id/interaction/:document_id
```
Comment (biernacki) [inline discovery_engine_core/web-api/src/main.rs:58]: change last segment to `body`

**Pull request closed without merging**